pandas/tests/indexes/test_any_index.py

```diff
@@ -27,9 +35,58 @@ def test_mutability(indices):
 
 
 def test_wrong_number_names(indices):
+    names = indices.nlevels * ["apple", "banana", "carrot"]
     with pytest.raises(ValueError, match="^Length"):
-        indices.names = ["apple", "banana", "carrot"]
+        indices.names = names
+
+
+class TestConversion:
+    def test_to_series(self, indices):
+        # assert that we are creating a copy of the index
+
+        ser = indices.to_series()
+        assert ser.values is not indices.values
+        assert ser.index is not indices
+        assert ser.name == indices.name
+
+    def test_to_series_with_arguments(self, indices):
+        # GH#18699
+
+        # index kwarg
+        ser = indices.to_series(index=indices)
+
+        assert ser.values is not indices.values
+        assert ser.index is indices
+        assert ser.name == indices.name
+
+        # name kwarg
+        ser = indices.to_series(name="__test")
+
+        assert ser.values is not indices.values
+        assert ser.index is not indices
+        assert ser.name != indices.name
+
+    def test_tolist_matches_list(self, indices):
+        assert indices.tolist() == list(indices)
+
+
+class TestRoundTrips:
+    def test_pickle_roundtrip(self, indices):
+        result = tm.round_trip_pickle(indices)
+        tm.assert_index_equal(result, indices)
+        if result.nlevels > 1:
+            # GH#8367 round-trip with timezone
+            assert indices.equal_levels(result)
+
+
+class TestIndexing:
+    def test_slice_keeps_name(self, indices):
+        assert indices.name == indices[1:].name
 
 
-def test_tolist_matches_list(indices):
-    assert indices.tolist() == list(indices)
+class TestRendering:
+    def test_str(self, indices):
```

Comment:
hmm, may need to rethink this test now testing over all indexes since "tuples": MultiIndex.from_tuples(zip(["foo", "bar", "baz"], [1, 2, 3])), so "foo" is already in str(indices)